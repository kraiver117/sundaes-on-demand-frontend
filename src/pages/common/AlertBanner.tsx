import React, { FC, RefAttributes } from "react";
import Alert from "react-bootstrap/Alert";
import { Color } from "react-bootstrap/esm/types";

type AlertBannerProps = {
  message?: string | null;
  variant?: Color | null;
} & RefAttributes<HTMLDivElement>;

export const AlertBanner: FC<AlertBannerProps> = ({ message, variant }) => {
  const alertMessage =
    message || "An unexpected error occurred. Please try again later.";

  const alertVariant = variant || "danger";

  return (
    <Alert variant={alertVariant} style={{ backgroundColor: "red" }}>
      {alertMessage}
    </Alert>
  );
};
