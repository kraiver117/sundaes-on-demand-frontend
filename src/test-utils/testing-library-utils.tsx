import { render, RenderOptions } from "@testing-library/react";
import { ReactElement, JSXElementConstructor } from "react";
import { OrderDetailsProvider } from "../contexts/OrderDetails";

const renderWithContext = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions
) =>
  render(ui, {
    wrapper: OrderDetailsProvider,
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { renderWithContext as render };
