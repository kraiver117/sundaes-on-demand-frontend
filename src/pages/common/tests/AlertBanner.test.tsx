import { render, screen } from "../../../test-utils/testing-library-utils";
import { AlertBanner } from "../AlertBanner";

test("Should render AlertBanner Correctly", () => {
  const { asFragment } = render(
    <AlertBanner message="Hi there" variant="success" />
  );

  expect(asFragment()).toMatchSnapshot();
});
