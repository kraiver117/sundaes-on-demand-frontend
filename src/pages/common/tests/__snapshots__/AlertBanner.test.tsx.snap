// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AlertBanner Correctly 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success show"
    role="alert"
    style="background-color: red;"
  >
    Hi there
  </div>
</DocumentFragment>
`;
